workflow/controller/operator.go

```diff
@@ -1069,11 +1069,12 @@ func (woc *wfOperationCtx) shouldPrintPodSpec(node wfv1.NodeStatus) bool {
 }
 
 // fails any suspended and pending nodes if the workflow deadline has passed
+// ensures that suspended and pending nodes marked failed after the reconciliation if the workflow shutting down
 func (woc *wfOperationCtx) failSuspendedAndPendingNodesAfterDeadlineOrShutdown() {
```

Comment:
This function was confusing because the name tells that pending nodes should be marked failed in the shutdown scenario, but it never happened because of this condition `(node.Phase == wfv1.NodePending && deadlineExceeded)` (as far as I understand). I changed it to `node.Phase == wfv1.NodePending` and it fixed the bug even without changes to exec_controll.go.
Finally, I decided to include updated condition in this PR in order to provide some failover logic in case some nodes weren't marked failed during pod reconciliation. Also it makes the function name not confusing.